### DISCUSSION TITLE
Some confusing language syntax  suggestions

### DISCUSSION THREAD

**Discussion by mrhuo** [vlang/v#11829] [Ideas]
### FIRST: I like vlang. I just want it to be better, It's that the perfect programming language I've been looking for. but it still has some confusing syntax (JUST I THINK).

`This issue I will keep updating, record some little little suggestions`

```v
struct App {
}

for i in input {
}

if value {
}

fn main() {
}
```
1. See above syntax or other not used, it use brace `{}`, it's has no problem, but see this.

```v
import os
import flag
import semver
import vxt
import java
```

**Can we use this:**

```v
import {
    os
    flag
    semver
    //...others
}
```
----
```v
const (
)
```

**Can we use this:**

```v
const {
}
```
`import` and `const`  are all defined keywords, it difference with function block, and no `fn` define.

**Comment by spytheman** [comment]
In earlier versions of V, you could do `import ( os sync )`.
import is deliberately the way it is now, because it makes tooling and searching for what imports what *much* easier.

**Comment by changrui** [comment]
`import os, io, encoding.csv, math.sin` is much better.

**Comment by kkharji** [comment]
`const {}` is way better than `const ()`. It's more consistent. But I guess this will never change

**Comment by mcastorina** [comment]
My thoughts:

* One `import` per line is only helpful for searching for imports; it may help dissuade importing many libraries into a single file.
* `const` lists can be huge, so using `()` is cleaner. Using a different syntax from blocks is a visual indicator that these values are not being defined at runtime. It also matches Go's syntax.

I could see an argument for doing `import ()`, but using `{}` for `const` and `import` doesn't look right to me.

**Comment by mrhuo** [comment]
All like this blocks if using `{}` , It will reduce a lot of things that need to be remembered and make coding more comfortable, is it?

**Comment by mrhuo** [comment]
2. I found some code like this:

```v
mut verbosity := fp.int_opt('verbosity', `v`, 'Verbosity level 1-3') or { 0 }
if ('-v' in os.args || 'verbosity' in os.args) && verbosity == 0 {
}
```

**What puzzles me is the two keywords `or` and `||`.**

if like this, is it better, clear?

```v
mut verbosity := fp.int_opt('verbosity', `v`, 'Verbosity level 1-3') || { 0 }
//or like this:
mut verbosity := fp.int_opt('verbosity', `v`, 'Verbosity level 1-3') || 0

if ('-v' in os.args || 'verbosity' in os.args) && verbosity == 0 {
}
```

**Keywords should the less the better, is it?**

**Comment by mcastorina** [comment]
I think it's important to keep those two separate because they are doing distinct actions:

* `or` is used for handling errors
* `||` is used for boolean expressions (e.g. `false || true`)

I agree that the less keywords the better, but not at the expense of readability. It would be harder to figure out whether `||` is being used for handling an error or for boolean algebra because now the keyword's meaning is context specific.

**Comment by mrhuo** [comment]
3. The documention sample code:

```v
fn main() {
	vweb.run<App>(8081)    <---- ok, invoke run function from moudule vweb
}
pub fn (mut app App) index() vweb.Result {     <---- ok, extention function from App
	message := 'Hello, world from Vweb!'
	return $vweb.html()    <---- but this, why start with $, is `vweb.html()` more better?
}
pub fn (app &App) init() {    <---- but this, while start with &, is `pub fn (app App) init()` more better? We can specify that a module has only one `init` function and `init_once ` function
}

pub fn (app &App) init_once() {    <---- are we should discard more prefix?
}
```

**Comment by spaceface777** [comment]
Also from the documentation: 


> If a function argument is immutable (like foo in the examples above) V can pass it either by value or by reference. The compiler will decide, and the developer doesn't need to think about it.
> 
> You no longer need to remember whether you should pass the struct by value or by reference.
> 
> You can ensure that the struct is always passed by reference by adding &:
> ```v
> struct Foo {
> 	abc int
> }
> 
> fn (foo &Foo) bar() {
> 	println(foo.abc)
> }
> ```

**Comment by mrhuo** [comment]
I think the cost of relearning a computer language is still very high. If everyone can understand a little basic grammar, they can easily switch from other languages. I think it is very meaningful.

**Comment by Terisback** [comment]
> but this, why start with $, is `vweb.html()` more better?

@mrhuo because it's compile time magic that takes template file and compiles it into v file and embeds it

**Comment by dumblob** [comment]
>I think the cost of relearning a computer language is still very high. If everyone can understand a little basic grammar, they can easily switch from other languages. I think it is very meaningful.

@mrhuo it's just that *V* documentation is not meant for beginner programmers. If it were, we'd have some step-by-step tutorial like Go has which would first tell you to use `pub fn (app App) init() { ... }` and then on the **very last** slide of the whole tutorial finally tell you there are also performance (`&` - e.g. `pub fn (app &App) init() { ... }`) and low-level dangerous things (`unsafe{}` etc.) for advanced programmers. That way it's easily digestable and you wouldn't be surprised by that any more.

In other words, your rants above are just about how one presents the language, but not about the language itself. Would you make a PR adding a tutorial for programming beginners?
